dataplane/linux/int_dataplane.go

```diff
@@ -1263,11 +1263,35 @@ func (d *InternalDataplane) setUpIptablesBPF() {
 			rulesConfig.RouteSource == "WorkloadIPs" {
 			log.Debug("Adding Wireguard iptables rule")
 			rawRules = append(rawRules, iptables.Rule{
-				Match: iptables.Match().Protocol("udp").
-					DestPorts(uint16(rulesConfig.WireguardListeningPort)).
-					NotSrcAddrType(iptables.AddrTypeLocal, false),
+				Match:  nil,
```

Comment:
Why is this logic duplicated here and in the rules renderer?  I think it should only be in the rules renderer.

Reply:
I thought the same logic is needed in both branches of this
[block](https://github.com/projectcalico/felix/pull/2820/files#diff-1f472efb904b55400c81c50c3b28ac6810f5ad14a3684ee8521f1f44371df772L1085-L1089)?

Reply:
Ah, yep, hadn't spotted this was on the BPF codepath.  You can still use the rule renderer to calculate that chain if it's the same.

That said, I'm not sure if this fix is right for eBPF mode; we try to avoid doing things in iptables when we're in eBPF mode.  If it works it might be a good stop gap until we port the logic over to the BPF code, though.

Reply:
Thanks for the ruleRenderer tip, I've changed this section to use it.